src/osd/OSD.cc

```diff
@@ -2512,6 +2513,25 @@ void OSD::create_logger()
 
   PerfCountersBuilder osd_plb(cct, "osd", l_osd_first, l_osd_last);
 
+  // Latency axis configuration for op histograms, values are in nanoseconds
+  PerfHistogramCommon::axis_config_d op_hist_x_axis_config{
+    "Latency (usec)",
+    PerfHistogramCommon::SCALE_LOG2, ///< Latency in logarithmic scale
+    0,                               ///< Start at 0
+    100000,                          ///< Quantization unit is 100usec
```

Comment:
> the axis scale is logarithmic here so that there's much higher granularity for lower values and much lower for the big ones.

But this prevents one from measuring maximal latency accurately enough. For instance, all values in [0.1 sec, 1 sec) end up in the same bin, it's impossible to distinguish between 0.11 sec and 0.99 sec, which kind of makes the data useless.

> I think it won't be a good fit here though since they don't support atomic/concurrent histograms.

The Java version does support thread safe (concurrent) and atomic versions (nobody bothered to port those to C yet).